server/service/client.go

```diff
@@ -309,6 +309,11 @@ func (c *Client) ApplyGroup(
 		if opts.DryRun {
 			logfn("[!] ignoring policies, dry run mode only supported for 'config' and 'team' specs\n")
 		} else {
+			// Policy names must be unique, return error if duplicate policy names are found
+			if policyName := specs.FirstDuplicatePolicySpecName(); policyName != "" {
```

Comment:
We should add the same check server side (to `ApplyPolicySpecs`):
https://github.com/fleetdm/fleet/blob/dda5a3138a5c8a955333ec19cbed4de907b26cb3/server/datastore/mysql/policies.go#L475

Reply:
Though... it might break some customers automations, am all ears here.

We already document that we ignore such fields: https://github.com/fleetdm/fleet/blob/dda5a3138a5c8a955333ec19cbed4de907b26cb3/docs/Contributing/API-for-contributors.md?plain=1#L970-L974

Reply:
Ah I didn't think of this, pinging product for guidance here @zhumo.  My opinion is that we should keep consistency in experience whether using fleetctl or the API.

Reply:
Aah great catch @lucasmrod ! Though I think this is very error-prone, as shown by a whole team of us not knowing that this was its normal/intended behaviour :D Very easy to miss this small note in the documentation, I think it would benefit from failing noisily instead of applying while ignoring those fields silently. But it is a breaking change, for sure.

Reply:
(at the very least, if we keep it like that, we should add a comment in the apply policies database method that points to this doc and mentions this is by design, because every time we see that ON DUPLICATE UPDATE clause we'll think that some fields are missing)

Reply:
I may be nitpicking here, but since it's a breaking change, should we save it for a minor release rather than a patch release?

Reply:
+1, I don't see a reason to release this on a patch release (e.g. v4.35.3). We only do patch releases for fixing high severity bugs.

Reply:
CC: @sharon-fdm

Reply:
Hey @mostlikelee, what is the tradeoff discovered here?

Reply:
Hey @mostlikelee, what is the tradeoff discovered here?

Reply:
Update:  Team change validation added to the `spec` endpoint to match other policy modifying endpoints